src/Stratis.Bitcoin.IntegrationTests/SegWitTests.cs

```diff
@@ -42,8 +43,8 @@ public void TestSegwit_MinedOnCore_ActivatedOn_StratisNode()
                 {
                     // generate 450 blocks, block 431 will be segwit activated.
                     coreRpc.Generate(450);
-
-                    TestHelper.WaitLoop(() => stratisNode.CreateRPCClient().GetBestBlockHash() == coreNode.CreateRPCClient().GetBestBlockHash());
+                    var cancellationToken = new CancellationTokenSource(TimeSpan.FromMinutes(2)).Token;
```

Comment:
Nice, I assume this explains why this test seems to fail randomly?

Reply:
yes, basically we have 2 different values for cancellation tokens, if you run it in debug it will get more time to check for the segwit block(with increased time you are going to pass), I think I will want to rewrite it not to relay on the wait loop to check if we got to block 450...